libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1245,10 +1246,16 @@ unsigned CompilerUtils::loadFromMemoryHelper(Type const& _type, bool _fromCallda
 		int shiftFactor = (32 - numBytes) * 8;
 		rightShiftNumberOnStack(shiftFactor);
 		if (leftAligned)
+		{
 			leftShiftNumberOnStack(shiftFactor);
+			cleanupNeeded = false;
+		}
+		else if (IntegerType const* intType = dynamic_cast<IntegerType const*>(&_type))
+			if (!intType->isSigned())
+				cleanupNeeded = false;
```

Comment:
Shouldn't this only be a special case for uint256?

Reply:
Right, this is in the non-32byte branch.

Ah, forgot it is BE and not LE. The code looks good.